src/App.js

```diff
@@ -1,21 +1,176 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from 'react'
+import './App.css'
+import Page from './Page.js'
+import LandingPage from './LandingPage.js'
+import LogInPage from './LogInPage.js'
+import firebase from './firebase.js'
+
+const LANDING = 'landing'
+const LOG_IN = 'login'
+const SIGN_UP = 'signup'
+const PROFILES = 'profiles'
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
+  constructor (props) {
+    super(props)
+    this.state = {
+      user: null,
+      userLocation: LANDING,
+      firstName: '',
+      lastName: '',
+      email: '',
+      phone: '',
+      password: '',
+      avatar: ''
+    }
+    this.toLogIn = this.toLogIn.bind(this)
+    this.toSignUp = this.toSignUp.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.logIn = this.logIn.bind(this)
+    this.setUser = this.setUser.bind(this)
+    this.clearUser = this.clearUser.bind(this)
+  }
+
+  toLogIn () {
+    this.setState({userLocation: LOG_IN})
+  }
+
+  logIn (e) {
+    e.preventDefault()
+    let userUID
+    const setUser = this.setUser
+    firebase.auth().signInWithEmailAndPassword(this.state.email, this.state.password)
+      .then(
+        firebase.auth().onAuthStateChanged(function (fbuser) {
```

Comment:
There's usually no good reason to use `function` instead of an arrow function. Replace with:
```
firebase.auth().onAuthStateChanged((fbuser) => {
...
```